### PR TITLE
🌱 Improve sort for v1beta2 conditions

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
make sure that well know v1beta2 conditions always appear in the same order

/area utils

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: util/conditions/v1beta2/sort.go [hunk 1]
```diff
@@ -16,22 +16,48 @@ limitations under the License.
...
+	return i.Type < j.Type
+}
+
+var first = map[string]int{
+	clusterv1.AvailableV1Beta2Condition: 0,
+	clusterv1.ReadyV1Beta2Condition:     1,
+}
+
+var last = map[string]int{
+	clusterv1.DeletingV1Beta2Condition: 0,
```
Comment (sbueringer) [inline util/conditions/v1beta2/sort.go:61]: WDYT about making deleting the very last condition?

I think we'll regularly look at the deleting condition and it's just perfect if it's the last condition when using `k get ... -o yaml`

File: util/conditions/v1beta2/sort.go [hunk 2]
```diff
@@ -16,22 +16,48 @@ limitations under the License.
...
 )
 
 // defaultSortLessFunc returns true if a condition is less than another with regards to the
 // order of conditions designed for convenience of the consumer, i.e. kubectl get.
-// According to this order the Available and the Ready condition always goes first, followed by all the other
-// conditions sorted by Type.
+// According to this order the Available and the Ready condition always goes first, Deleting and Paused always goes last,
+// and all the other conditions are sorted by Type.
 func defaultSortLessFunc(i, j metav1.Condition) bool {
-	return (i.Type == AvailableCondition || (i.Type == ReadyCondition || i.Type < j.Type) && j.Type != ReadyCondition) && j.Type != AvailableCondition
```
Comment (sbueringer) [inline util/conditions/v1beta2/sort.go:36]: Disappointed this is not a one-liner anymore :joy:

**Comment by sbueringer** [comment]
Thank you!

/lgtm
/approve

**Pull request merged**